doc/develop/west/alias.rst

```diff
@@ -0,0 +1,56 @@
+.. _west-aliases:
+
+West aliases
+############
+
+West allows to add alias commands to the local, global or system configuration files.
+These aliases make it easy to add shortcuts for frequently used, or hard to memorize
+commands for ease of development.
+
+Similar to how ``git`` aliases work, the alias command is replaced with the alias'
+full text and parsed as a new shell argument list. This enables adding argument
+parameters as they were passed to the original command. Spaces are considered
+argument separators, use proper escaping if arguments shouldn't be split.
+
+To add a new alias simply call the ``west config`` command:
+
+.. code-block:: console
+
+   west config alias.command "other-command --with-arg=\"some value\""
```

Comment:
Alternate quotes instead of using backslash to keep the example more readable.

Also, how about a more realistic example?

```suggestion
   west config alias.mylist "list -f '{name} {revision}'"
```